platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,193 @@
+  }
+
+  @Test
+  public void parseNullMetadata() throws Exception {
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(null, bindingSet);
+
+    assertThat(testMap, is(equalTo(EMPTY_MAP)));
+  }
+
+  @Test
+  public void testentityID2s() throws Exception {
+    assertThat(entityID, is(equalTo(ENTITY_ID)));
+    assertThat(entityID2, is(equalTo(ENTITY_ID_2)));
+  }
+
+  @Test
+  public void testSigningCertificates() {
+    assertThat(testInfo.getSigningCertificate(), is(equalTo(certificate)));
+    assertThat(testInfo2.getSigningCertificate(), is(equalTo(certificate)));
+  }
+
+  @Test
+  public void testEncryptionCertficates() {
+    assertThat(testInfo.getEncryptionCertificate(), is(equalTo(certificate)));
+    assertThat(testInfo2.getEncryptionCertificate(), is(equalTo(certificate)));
+  }
+
+  @Test
+  public void testRedirectLogoutUrls() {
+    assertThat(
+        testInfo.getLogoutService(Binding.HTTP_REDIRECT).getUrl(),
+        is(equalTo(REDIRECT_LOGOUT_URL)));
+    assertThat(
+        testInfo2.getLogoutService(Binding.HTTP_REDIRECT).getUrl(),
+        is(equalTo(REDIRECT_LOGOUT_URL)));
+  }
+
+  @Test
+  public void testPostLogoutUrls() {
+    assertThat(
+        testInfo2.getLogoutService(Binding.HTTP_POST).getUrl(), is(equalTo(POST_LOGOUT_URL)));
+    assertThat(testInfo.getLogoutService(Binding.HTTP_POST).getUrl(), is(equalTo(POST_LOGOUT_URL)));
+  }
+
+  @Test
+  public void testRedirectAssertionUrls() {
+    assertThat(
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_REDIRECT, null).getUrl(),
+        is(equalTo(REDIRECT_ACS_URL)));
+    assertThat(
+        testInfo2.getAssertionConsumerService(null, Binding.HTTP_REDIRECT, null).getUrl(),
+        is(equalTo(REDIRECT_ACS_URL)));
+  }
+
+  @Test
+  public void testPostAssertionUrls() {
+    assertThat(
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_POST, null).getUrl(),
+        is(equalTo(POST_ACS_URL)));
+    assertThat(
+        testInfo2.getAssertionConsumerService(null, Binding.HTTP_POST, null).getUrl(),
+        is(equalTo(POST_ACS_URL)));
+  }
+
+  @Test
+  public void testSoapAssertionUrl() {
+    assertThat(
+        testInfo2.getAssertionConsumerService(null, Binding.SOAP, null).getUrl(),
+        is(equalTo(SOAP_ACS_URL)));
+  }
+
+  @Test
+  public void testNoSoapAssertionUrl() {
+    assertThat(
+        String.format(
+            "Expected to not find a SOAP binding (url: %s), and for the method to return the default Post binding (url: %s) instead",
+            SOAP_ACS_URL, POST_ACS_URL),
+        testInfo.getAssertionConsumerService(null, Binding.SOAP, null).getUrl(),
+        is(equalTo(POST_ACS_URL)));
+  }
+
+  @Test
+  public void testArtifactAssertionUrl() {
+    assertThat(
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_REDIRECT, null).getUrl(),
+        is(equalTo(REDIRECT_ACS_URL)));
+  }
+
+  @Test
+  public void testNoRedirectAssertionUrl() {
```

Comment:
✏️ I think that the `AssertionUrl` and `SoapAssertionUrl` tests should be combined to two total tests. That way each test is asserting on the field for both info objects, just like all the other tests.

I see why you broke these two properties out since they're different between the `EntityInformation` objects. But it's inconsistent and makes the test names sound weird.